metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -358,27 +351,75 @@ func CreateStackdriverSink(uri *url.URL) (core.DataSink, error) {
 		}
 	}
 
-	if err := gce_util.EnsureOnGCE(); err != nil {
-		return nil, err
+	var projectId, heapsterZone string
+	// Cluster name and location are required when useNewResourceModel is true.
+	var clusterName, clusterLocation string
+
+	if len(opts["cluster_name"]) >= 1 {
+		clusterName = opts["cluster_name"][0]
 	}
 
-	// Detect project ID
-	projectId, err := gce.ProjectID()
-	if err != nil {
-		return nil, err
+	if len(opts["cluster_location"]) >= 1 {
+		clusterLocation = opts["cluster_location"][0]
 	}
 
-	// Detect zone for old resource model
-	heapsterZone, err := gce.Zone()
-	if err != nil {
-		return nil, err
+	if gce.OnGCE() {
+		// Detect project ID
+		projectId, err = gce.ProjectID()
+		if err != nil {
+			return nil, err
+		}
+
+		// Detect zone for old resource model
+		heapsterZone, err = gce.Zone()
+		if err != nil {
+			return nil, err
+		}
+
+		if useNewResourceModel {
+			if clusterName == "" {
+				glog.Info("An empty cluster name has been provided, checking the GCE Metadata Server to try to auto-detect.")
+
+				clusterName_, err := gce.InstanceAttributeValue("cluster-name")
```

Comment:
There is no sense in introducing additional variable.
InstanceAttributeValue is going to return empty string in case of error. So this could be simplified to:
clusterName, err = gce.InstanceAttributeValue("cluster-name")